servers/physics_2d/space_2d_sw.cpp

```diff
@@ -989,7 +989,7 @@ bool Space2DSW::test_body_motion(Body2DSW *p_body, const Transform2D &p_from, co
 			if (stuck) {
 
 				safe = 0;
-				unsafe = 0;
+				unsafe = 1.0 / (1 << 8);
```

Comment:
There should probably be a comment above this line to explain why this is done exactly. (Feel free to use `GH-38529` as a reference to this pull request.)

Reply:
How about:
> // Set unsafe to the minimum after eight steps = 1/2^8.

Reply:
I think code comments like these should be more about the *why* than the *how*. If I read that comment, I have no idea why it's done exactly :wink:

Reply:
Are you talking about why the unsafe is being set to the minimum, why the minimum is 1/2^8 or why 1/(1<<8) = 1/2^8?

Reply:
Maybe something like this?

```cpp
// When a KinematicBody is stuck, setting the unsafe proportion to the
// minimum instead of zero ensures that collision information is extracted // even if the penetration causing the body to be stuck is less than the
// `test_motion_min_contact_depth`.
```

Reply:
Mmm. I think that is excessive. Code shouldn't document why a change was made. It makes it look like setting the value to 1/2^8 is unusual, which it isn't. It is simply the value that unsafe would have been set to if this loop had been allowed to run:
https://github.com/godotengine/godot/blob/e253451a5b01926a32aa52bf2e78150f0f9f4bd5/servers/physics_2d/space_2d_sw.cpp#L948